### PR TITLE
Various documentation fixes

### PR DESCRIPTION
**Pull request by poettering**
Let's close a bunch of issues.

### CODE REVIEW

File: man/systemd-cryptenroll.xml [hunk 1]
```diff
@@ -235,6 +235,30 @@
...
+      <listitem><para>The opposite is not guaranteed however: it might not be possible to unlock volumes with
+      enrollments done with a newer version of <command>systemd-cryptenroll</command> with an older version
+      of <command>systemd-cryptsetup</command>.</para></listitem>
+    </itemizedlist>
+
+    <para>That said, it is generally recommended to use matching versions of
+    <command>systemd-cryptenroll</command> and <command>systemd-cryptsetup</command>, since this is best
+    tested and supported.</para>
+
+    <para>It might be advisable to re-enroll existing enrollments regularly to take benefit of newer security
```
Comment (bluca) [inline man/systemd-cryptenroll.xml]: Constant re-enrollment is a pain that we want to get away from, so I'd drop this para

**Review by bluca — APPROVED** [review]

File: man/systemd-cryptenroll.xml (hunk 1 above)
Comment (poettering) [reply man/systemd-cryptenroll.xml]: the thing is that early tpm enrollments didn't make use of param encryption, pinning and salting. You only get those if you re-enroll. I'll reword it slightly suggesting to do it only after updates.

Comment (bluca) [reply man/systemd-cryptenroll.xml]: Not just after any upgrade, but after upgrades that introduce new features that are desired

**Comment by poettering** [comment]
toned down the compat sentence. merging now, not waiting for CI, this is only docs after all.

**Pull request merged**